dali/core/mm/async_pool_test.cu

```diff
@@ -351,5 +350,103 @@ TEST(MMAsyncPool, CrossStreamWithHogs) {
   upstream.check_leaks();
 }
 
+#if DALI_USE_CUDA_VM_MAP
+
+TEST(MM_VMAsyncPool, MultiThreadedSingleStreamRandom) {
+  if (!cuvm::IsSupported())
+    GTEST_SKIP() << "Virtual memory management API is not supported on this machine.";
+
+  CUDAStream stream = CUDAStream::Create(true);
+  {
+    vector<block> blocks;
+    std::mutex mtx;
+
+    async_pool_resource<memory_kind::device, cuda_vm_resource> pool;
+
+    vector<std::thread> threads;
+
+    for (int t = 0; t < 10; t++) {
+      threads.push_back(std::thread([&]() {
+        AsyncPoolTest(pool, blocks, mtx, stream);
+      }));
+    }
+    for (auto &t : threads)
+      t.join();
+  }
+}
+
+TEST(MM_VMAsyncPool, MultiThreadedMultiStreamRandom) {
+  if (!cuvm::IsSupported())
+    GTEST_SKIP() << "Virtual memory management API is not supported on this machine.";
+
+  async_pool_resource<memory_kind::device, cuda_vm_resource> pool;
+
+  vector<std::thread> threads;
+
+  for (int t = 0; t < 10; t++) {
+    threads.push_back(std::thread([&]() {
+      CUDAStream stream = CUDAStream::Create(true);
+      vector<block> blocks;
+      detail::dummy_lock mtx;
+      AsyncPoolTest(pool, blocks, mtx, stream);
+      CUDA_CALL(cudaStreamSynchronize(stream));
+    }));
+  }
+  for (auto &t : threads)
+    t.join();
+}
+
+TEST(MM_VMAsyncPool, MultiStreamRandomWithGPUHogs) {
+  if (!cuvm::IsSupported())
+    GTEST_SKIP() << "Virtual memory management API is not supported on this machine.";
+
+  async_pool_resource<memory_kind::device, cuda_vm_resource> pool;
+
+  vector<std::thread> threads;
+
+  for (int t = 0; t < 10; t++) {
+    threads.push_back(std::thread([&]() {
+      // 0-th thread uses null stream, which triggers non-async API usage
+      CUDAStream stream = t ? CUDAStream::Create(true) : CUDAStream();
+      vector<block> blocks;
+      detail::dummy_lock mtx;
+      AsyncPoolTest(pool, blocks, mtx, stream, 20000, true);
+      CUDA_CALL(cudaStreamSynchronize(stream));
+    }));
+  }
+  for (auto &t : threads)
+    t.join();
+}
+
+
+
+TEST(MM_VMAsyncPool, CrossStreamWithHogs) {
```

Comment:
Why not add `TEST(MM_VMAsyncPool, CrossStream) {` as well?

Reply:
I can add it.